sync/src/synchronizer/block_fetcher.rs

```diff
@@ -146,16 +146,17 @@ impl BlockFetcher {
             for _ in 0..span {
                 let parent_hash = header.parent_hash();
                 let hash = header.hash();
-                // NOTE: Filtering `BLOCK_STORED` but not `BLOCK_RECEIVED`, is for avoiding
-                // stopping synchronization even when orphan_pool maintains dirty items by bugs.
-                let stored = self
-                    .active_chain
-                    .contains_block_status(&hash, BlockStatus::BLOCK_STORED);
-                if stored {
+
+                let status = self.active_chain.get_block_status(&hash);
+                if status == BlockStatus::BLOCK_STORED {
                     // If the block is stored, its ancestor must on store
                     // So we can skip the search of this space directly
                     break;
-                } else if inflight.insert(self.peer, hash) {
+                } else if self.ibd.into() && status.contains(BlockStatus::BLOCK_RECEIVED) {
```

Comment:
It also redoes the download when not in IBD and the block has been received.

Reply:
Yes, this is currently waiting for asynchronous validation, otherwise that can't add this restriction

Reply:
I think the issue of this branch is that the comment is all about why 'ibd' is in the condition, but there's no comment to explain why this branch body is empty.